DockerfileLeanFoundation

```diff
@@ -44,25 +40,40 @@ RUN echo "deb http://download.mono-project.com/repo/ubuntu stable-xenial/snapsho
     apt-get update && apt-get install -y binutils mono-complete ca-certificates-mono mono-vbnc nuget referenceassemblies-pcl && \
     apt-get install -y fsharp && rm -rf /var/lib/apt/lists/* /tmp/*
 
-# Install miniconda and supported third party python packages
+# Install miniconda
+ENV CONDA="Miniconda3-4.5.12-Linux-x86_64.sh"
 ENV PATH="/opt/miniconda3/bin:${PATH}"
-RUN wget https://cdn.quantconnect.com/miniconda/Miniconda3-4.3.31-Linux-x86_64.sh && \
-    bash Miniconda3-4.3.31-Linux-x86_64.sh -b -p /opt/miniconda3 && \
-    rm -rf Miniconda3-latest-Linux-x86_64.sh && \
-    ln -s /opt/miniconda3/lib/libpython3.6m.so /usr/lib/libpython3.6m.so && \
-    conda update -y conda pip && \
-    conda install -y python=3.6.7 && \
-    conda install -y numpy=1.14.5 && \
-    conda install -y pandas=0.23.4 && \
-    conda install -y jsonschema lxml plotly seaborn quandl && \
-    conda install -y blaze cython cvxopt keras nltk tensorflow theano && \
-    conda install -y pytorch torchvision -c pytorch && \
-    conda install -y -c conda-forge rauth && \
-    pip install --upgrade pip && \
-    pip install arch copulalib cvxpy deap gym pykalman pyro-ppl  && \
-    pip install sklearn statistics tensorforce xgboost && \
-    pip install QuantLib-Python && \
-    conda clean -y --all
+RUN wget https://cdn.quantconnect.com/miniconda/${CONDA} && \
+    bash ${CONDA} -b -p /opt/miniconda3 && rm -rf ${CONDA} && \
+    ln -s /opt/miniconda3/lib/libpython3.6m.so /usr/lib/libpython3.6m.so
+
+# Install supported third party python packages
+RUN conda update -y conda pip
+RUN conda install -y python=3.6.8
+RUN conda install -y msgpack-python=0.5.6
+RUN conda install -y numpy=1.16.1
+RUN conda install -y pandas=0.23.4
+RUN conda install -y wrapt=1.10.11
+RUN conda install -y dill=0.2.9
+RUN conda install -y jsonschema lxml
+RUN conda install -y blaze cython cvxopt keras nltk tensorflow
```

Comment:
Suggest all of these use specifically requested versions or we'll pay for it later :D